src/helpers.js

```diff
@@ -155,3 +155,211 @@ export const isRectSelected = rect => selection => {
+//    ['a3', 'b3', 'c3', 'd3'],
+//  ]
+// ```
+export const transpose = array => {
+  return array[0].map((_, i) => {
+    return array.map(column => column[i]);
+  });
+};
+
+// :: (tableNode: Node) -> Array<Node>
+// This function will transform the table node
+// into a matrix of rows and columns respecting merged cells,
+// for example this table will be convert to the below:
+//
+//  ____________________________
+// |      |      |             |
+// |  A1  |  B1  |     C1      |
+// |______|______|______ ______|
+// |      |             |      |
+// |  A2  |     B2      |      |
+// |______|______ ______|      |
+// |      |      |      |  D1  |
+// |  A3  |  B3  |  C2  |      |
+// |______|______|______|______|
+//
+//
+// array = [
+//   [A1, B1, C1, null],
+//   [A2, B2, null, D1],
+//   [A3. B3, C2, null],
+// ]
+export const convertTableNodeToArrayOfRows = tableNode => {
+  const map = TableMap.get(tableNode);
+  const rows = [];
+  for (let rowIndex = 0; rowIndex < map.height; rowIndex++) {
+    const rowCells = [];
+    const seen = {};
+
+    for (let colIndex = 0; colIndex < map.width; colIndex++) {
+      const cellPos = map.map[rowIndex * map.width + colIndex];
+      const cell = tableNode.nodeAt(cellPos);
+      const rect = map.findCell(cellPos);
+      if (seen[cellPos] || rect.top !== rowIndex) {
+        rowCells.push(null);
+        continue;
+      }
+      seen[cellPos] = true;
+
+      rowCells.push(cell);
+    }
+
+    rows.push(rowCells);
+  }
+
+  return rows;
+};
+
+// :: (tableNode: Node, tableArray: Array<Node>) -> Node
+// This function will transform a matrix of nodes
+// into table node respecting merged cells and rows configurations,
+// for example this array will be convert to the table below:
+//
+// array = [
+//   [A1, B1, C1, null],
+//   [A2, B2, null, D1],
+//   [A3. B3, C2, null],
+// ]
+//  ____________________________
+// |      |      |             |
+// |  A1  |  B1  |     C1      |
+// |______|______|______ ______|
+// |      |             |      |
+// |  A2  |     B2      |      |
+// |______|______ ______|      |
+// |      |      |      |  D1  |
+// |  A3  |  B3  |  C2  |      |
+// |______|______|______|______|
+//
+export const convertArrayOfRowsToTableNode = (tableNode, arrayOfNodes) => {
+  const rowsPM = [];
+  const map = TableMap.get(tableNode);
+  for (let rowIndex = 0; rowIndex < map.height; rowIndex++) {
+    const row = tableNode.child(rowIndex);
+    const rowCells = [];
+
+    for (let colIndex = 0; colIndex < map.width; colIndex++) {
+      if (!arrayOfNodes[rowIndex][colIndex]) {
+        continue;
+      }
+      const cellPos = map.map[rowIndex * map.width + colIndex];
+
+      const cell = arrayOfNodes[rowIndex][colIndex];
+      const cellData = [
+        Object.assign({}, cell.attrs),
+        cell.content,
+        cell.marks,
+      ];
+
+      const oldCell = tableNode.nodeAt(cellPos);
+      const newCell = oldCell.type.createChecked.apply(oldCell.type, cellData);
```

Comment:
What's the benefit of creating an array and using `apply`? You are not using that array anywhere else.

Reply:
I need to replace the spread operation to keep builddocs works on that time it was complain every time that I was using spread operations. I replaced on this commit ae9fa6e9e9aa3282113048f596405b5cd54af3bc

Reply:
My point was don't create the array, do following instead:
```js
const newCell = oldCell.type.createChecked(Object.assign({}, cell.attrs), cell.content, cell.marks)
```

Reply:
🥇 much better, I didn't thought on this.